h/api/nipsa/search.py

```diff
@@ -0,0 +1,83 @@
+import copy
+
+
+def nipsa_filter(query, userid=None):
```

Comment:
I really think that spreading our knowledge and logic about what the structure of Es queries is and how to construct them in different modules throughout the app is a pretty big negative..